File: app/src/main/java/eu/spanhel/heating/RoomHeatingActivity.java
package eu.spanhel.heating;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.Spinner;

import java.util.Locale;

public class RoomHeatingActivity extends AppCompatActivity {
    double newTemperature;
    EditText etTemper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_room_heating);

        Intent intent = getIntent();

        CRoomHeatingParams heatingParams = (CRoomHeatingParams) intent.getSerializableExtra("HeatingParams");

        etTemper = findViewById(R.id.editTemper);

        Spinner roomsSpinner = findViewById(R.id.room_spinner);
        roomsSpinner.setAdapter(null);

        if (heatingParams != null) {
            ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, heatingParams.rooms);
            roomsSpinner.setAdapter(adapter);
            newTemperature = heatingParams.setTemperature;
            etTemper.setText(String.format(Locale.ENGLISH, "%.1f", newTemperature));

            for (int i = 0; i < heatingParams.rooms.size(); i++) {
                if (heatingParams.rooms.get(i).equals(heatingParams.selectedRoom)) {
                    roomsSpinner.setSelection(i);
                    break;
                }
            }
        }
    }

    public void btnMinusOnClick(View view) {
        newTemperature -= 0.5;
        etTemper.setText(String.format(Locale.ENGLISH, "%.1f", newTemperature));
    }

    public void btnPlusOnClick(View view) {
        newTemperature += 0.5;
        etTemper.setText(String.format(Locale.ENGLISH, "%.1f", newTemperature));
    }

    public void btnSaveOnClick(View view) {
        Intent data = new Intent();
        Spinner roomsSpinner = findViewById(R.id.room_spinner);
        data.putExtra("room", roomsSpinner.getSelectedItem().toString());
        data.putExtra("temperature", newTemperature);
        setResult(RESULT_OK, data);
        finish();
    }

    public void btnBackClick(View view) {
        setResult(RESULT_CANCELED);
        finish();
    }
}

File: app/src/main/java/eu/spanhel/heating/RoomHeatingActivity.kt
package eu.spanhel.heating

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Spinner
import kotlinx.android.synthetic.main.activity_room_heating.*

class RoomHeatingActivity : AppCompatActivity() {

    private var listOfRooms = arrayOf("Ložnice", "Pokojík", "Kotelna", "Obývák", "Kuchyň")
    var spinner:Spinner? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_room_heating)

        spinner = this.room_spinner

        var aa = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, listOfRooms)
        aa.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner!!.adapter = aa

    }

}
